### PR TITLE
Rework of home page

### LINKED ISSUE #13382: Rework main documentation page
I find the main documentation page not quite welcoming and helpful for users. Therefore, I propose to rework it.

Comments on the current sections:

- Introductory text: Conceptually ok, but too wordy and maybe too much detail at some points.
- Installation: No Information other than a link. Could be included in some top-Level menu. Not also that many people, will just use the preinstalled version in their OS or anaconda or just pip install. Detailed installation instructions are more like an advanced topic nowerdays.
- Documentation: Is it really necessary to confront the user with various versions of the docs in such a prominent place? Some sort of version dropdown menu on the side would be better.
- Need help: Rather confusing than helpful with that many options and links. Also very wordy, e.g. the sentence on the search tool could be left out completely. We have the search box in a prominent place and people know how to use these things 😄 

Things I’m missing:

- better structure: There’s no kind of TOC. You have to scroll down to see which topics are actually on the page.
- a clear reference “Getting started” for new users (well, I think the appropriate content for this is also missing, but that’s another topic.

IMO https://seaborn.pydata.org is an example of a good main page. - Not that we have to do exactly the same, but it shows how one can do it in a clean, informative and compact way.

### PR DESCRIPTION
**Pull request by dstansby**
This is a fairly major re-work of the `matplotlib.org` landing page, primarily to vastly reduce the amount of information, but still provide easy to follow links to everything currently on the home page:
Screenshot 2019-08-28 at 09 47 40

In particular, the following sections that were text on the home page have moved:

- **Documentation** > `users/previous_versions.rst`
- **Other learning resources** > `users/external_resources.rst`
- **Need help** > `users/getting_help.rst`
- **Toolkits** > already has a page at `api/toolkits/index.rst`
- **Third party packages** > already has a page at `thirdpartypackages/index.rst`
- **Citing Matplotlib** > `users/citing.rst`

Other moves:

- **Installing FAQ** moved to the bottom of **Installing** page

todo
----
- Work out where the best place to put "Other versions" is
- Work out what text/links to go under each heading

As always (and especially here) feedback very welcome - I am by no means an expert at website design!

Fixes #13382

### CODE REVIEW

**Comment by dstansby** [comment]
You mean for backends etc? That's a good point, maybe it's best just to link to the `Installing` page instead of trying to provide a one-liner.

**Comment by timhoffm** [comment]
:tada: :tada: :tada: This is much more readable and pleasant. Still I think Numfocus takes too much room for a front page.

**Comment by ImportanceOfBeingErnest** [comment]
Depriving users of any kind of information on the landing page is pretty much en vogue nowadays; but honestly, I doubt there is any objective reason for that, so maybe matplotlib shouldn't follow that tendency? 


Anyways, what I'm missing in this current proposal is

* **Need help** We already have too many confused users asking questions in the wrong places. This should be prominently placed on the home page.
* Single-click access to the 2.x documentation. 
* **Getting started**, i.e. where should I click first when having no clue where to click first.
* **Function overview**, i.e. where should I start navigating when looking for some function or method. (Currently I could click on modules/matplotlib.axes/scroll down, or I could click on API / OO-interface / Axes; but both are kind of hidden if I don't know that I need to find the axes'  method, or if I don't know what an API is; so some guidance is necessary)

**Comment by dstansby** [comment]
Thanks for the feedback/suggestions! I've updated the top post with a new version that I think addresses some of the feedback; basically the home page is split into 6 sections (inspired by [astropy](https://www.astropy.org/) this time):

- **Install**
- **Learn**
- **Get help**
- **Other versions** (not sure this fits in too well, I think this should maybe be a 'floating' box somewhere?)
- **Get involved** (ie. contribute, maybe there's a better title for this?)
- **Support**

Thoughts on this approach?

**Comment by anntzer** [comment]
I think if you're going to remove most of the text (I don't have a very strong opinion there), the section titles should be made smaller -- right now they're huge, and bump a bit visually into one another.
Also, I would be sad to see the sentence "Matplotlib tries to make easy things easy and hard things possible." go away -- I think it defines Matplotlib's philosophy quite well.

**Comment by timhoffm** [comment]
I think @dorafc has a very good point concerning the strategic understanding. We should do some user stories: Who is coming to the website, what is their prior knowledge, what are they interested in?

Each of the user groups should find an entry point for their topic on the front page. Verbose content should have a very good reason to be up front. It helps the one user group that's interested in that, but clutters the page for all others and makes it harder for them to find their entry point.

**Comment by dorafc** [comment]
Here's how I'd recommend moving forward on improving the homepage (and the website in general) -

1. Identify the different users groups who use the Matplotlib website (ie. devs learning mpl)
2. Make a best guess at what each groups hopes to get from the site (ie. how to get mpl installed)
3. Other goals that you as the site stakeholders want the site to achieve (ie. learn that mpl is great!)

Once you've identified these rough metrics, it's a lot easier to make informed decisions about what content belongs on the site.

Tom and I have had many conversations about this (side effect of me seeing him a lot and me being a professional web person), and I already have a bunch of random scraps and wireframes about a better mpl website. I'm happy to set up a time to discuss the goals for an improved mpl website.

**Comment by jklymak** [comment]
Can I suggest we make those lists on a collaborative document of some sort and then have a discussion on things that aren’t agreed upon?  

As an experienced user I almost never come to the website unless google leads me here (which it does frequently). So we should really think about who benefits the most from organization of the material and aim at those audiences versus folks who would usually just take random access to it.
